packages/theme-check-common/src/checks/valid-schema-name/index.ts

```diff
@@ -0,0 +1,82 @@
+import { JSONNode, LiquidCheckDefinition, Severity, SourceCodeType } from '../../types';
+import { toJSONAST } from '../../to-source-code';
+import { visit } from '../../visitor';
+import { LiteralNode } from 'json-to-ast';
+import { ValueNode } from 'json-to-ast';
+
+const MAX_SCHEMA_NAME_LENGTH = 25;
+const ROOT_NODE_ANCESTORS_COUNT = 1;
+
+export const ValidSchemaName: LiquidCheckDefinition = {
+  meta: {
+    code: 'ValidSchemaName',
+    name: 'Enforce valid schema name',
+    docs: {
+      description: 'This check is aimed at ensuring a valid schema name.',
+      recommended: true,
+      url: 'https://shopify.dev/docs/storefronts/thesadmes/tools/theme-check/checks/valid-schema-name',
+    },
+    type: SourceCodeType.LiquidHtml,
+    severity: Severity.ERROR,
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    return {
+      async LiquidRawTag(node) {
+        if (node.name !== 'schema' || node.body.kind !== 'json') {
+          return;
+        }
+
+        const jsonSchemaString = node.source.slice(
+          node.blockStartPosition.end,
+          node.blockEndPosition.start,
+        );
+
+        const jsonSchemaAst = toJSONAST(jsonSchemaString);
+        visit<SourceCodeType.JSON, void>(jsonSchemaAst as JSONNode, {
+          Property(nameNode, ancestors) {
+            if (
+              nameNode.key.value === 'name' &&
+              ancestors.length === ROOT_NODE_ANCESTORS_COUNT &&
+              isLiteralNode(nameNode.value)
+            ) {
+              const name = getLiteralValue(nameNode.value);
+
+              // We will be handling the translation length in a follow up
+              if (!name.startsWith('t:') && name.length > MAX_SCHEMA_NAME_LENGTH) {
+                const startIndex = node.blockStartPosition.end + getLiteralLocStart(nameNode.value);
+                const endIndex = node.blockStartPosition.end + getLiteralLocEnd(nameNode.value);
+                context.report({
+                  message: 'Schema name is too long (max 25 characters)',
+                  startIndex: startIndex,
+                  endIndex: endIndex,
+                });
+              }
```

Comment:
I was going to recommend to just use the [JSON schema `min/maxLength` rules](https://json-schema.org/understanding-json-schema/reference/string#length). But I could see how we'd want to centralize this logic given we can't really do `t:` with just JSON schemas...

![image](https://github.com/user-attachments/assets/2161c263-e9da-4b5c-9d15-b45f391d54e9)

What's the rule we want for `t:`? Same thing but query the file? 

```jsonc
{
  "properties": {
    "name": {
      "type": "string",
      "description": "The name attribute determines the block title that's shown in the theme editor.",
      "markdownDescription": "The `name` attribute determines the block title that's shown in the theme editor.\n\n---\n\n[Shopify reference](https://shopify.dev/docs/themes/architecture/blocks/theme-blocks/schema#name)",
      "allOf": [
        {
          "if": { "pattern": "^t:" },
          "else": {
            "maxLength": 25
          }
        }
      ]
    },
  }
}
```